### ISSUE TITLE
Infinity PP

### ISSUE THREAD

**Issue by Fleepy** [Piotrekol/StreamCompanion#77]
I just finished playing a map, with 57 PP, and streamcompanion is telling me i have infinity PP.. As much as id love to have infinity performance points, this must be a bug.

https://i.imgur.com/RkOZa7M.png

**Comment by Piotrekol** [comment]
Seems to be only happening on only one diff from this set https://osu.ppy.sh/beatmapsets/323751#osu/914237

Will check later what is root cause.

**Comment by Fleepy** [comment]
Its also doing it on multiple beatmaps aswell. I dont want to list all of them because I feel like it doesn't matter which beatmap it is, and that it has something to do with how SC reads it.

**Comment by Piotrekol** [comment]
I never said it doesn't occur on other beatmaps, and now that I know cause - it can occur on any map(not specific to osu! mode either).

tl;dr: timing issue.
This is caused by getting anything other than 300hit on your first note - there is currently a chance that time read by memory reader will be earlier than time your note appears. Doesn't sound too bad by itself, but when used in pp calculator it results in -Infinity in accuracy... breaking everything else in the process.

I've added some timing gates & shuffled some code around so it will occur way less frequently, and even when it happens - it'll be silenced(NaN internally, but shown as 0) and be able to recover almost immediately(next memory read cycle, few ms at worst), so most likely you won't be even able to see it.
